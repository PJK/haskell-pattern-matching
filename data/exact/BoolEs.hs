module BoolEs where

-- Not exhaustive
f :: Bool -> Bool -> Int
f x y
    | x && y = 1
    | x || y = 2
