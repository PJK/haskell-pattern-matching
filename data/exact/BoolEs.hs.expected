{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "f",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "WildcardPattern",
                            "contents": []
                        },
                        {
                            "tag": "WildcardPattern",
                            "contents": []
                        }
                    ]
                ]
            }
        ]
    ]
}
